shub/config.py

```diff
@@ -164,15 +171,47 @@ def get_version(self):
         elif self.version:
             return str(self.version)
 
+    def get_stack(self, target):
+        return self.stacks.get(target)
+
     def get_target(self, target, auth_required=True):
         """Return (project_id, endpoint, apikey) for given target."""
-        return (
-            self.get_project_id(target),
-            self.get_endpoint(target),
-            self.get_apikey(target, required=auth_required),
+        return Target(
+            name=target,
+            project_id=self.get_project_id(target),
+            endpoint=self.get_endpoint(target),
+            apikey=self.get_apikey(target, required=auth_required),
+            stack=self.get_stack(target),
+            requirements_file=self.requirements_file,
         )
 
 
+class Target(object):
+
```

Comment:
Damn, we (well, okay, I) just introduced this API and it already falls apart.

Yeah, introducing `get_target_conf()` or similar sounds like the cleanest solution. I can't resist mentioning that especially now that we wouldn't have to extend `__getitem__` or `__eq__`, I would definitely prefer `namedtuple`, if only for the free `__repr__` and keyword args for `__init__` we get with it. However, it's a small detail, this is the last time I mention it and if you insist on writing the class code out instead of using the factory I will not fight you. :P